### PR TITLE
TextBox text style improvements

### PR DESCRIPTION
**Pull request by IceSentry**
* Use colour style for the text
* Use gray colour for the placeholder text
* Add more color constants

I'm not sure if it works on main because it doesn't compile for me, but it did work on the update-bevy branch.

Here's how it looks
image

### CODE REVIEW

File: kayak_core/src/color.rs [hunk 1]
```diff
@@ -34,6 +34,38 @@ impl Color {
...
+    };
+
+    /// The blue color.
+    pub const BLUE: Color = Color {
+        r: 0.0,
+        g: 0.0,
+        b: 1.0,
+        a: 1.0,
+    };
+
```
Comment (MrGVSV) [inline kayak_core/src/color.rs:68]: This might need to change due to your other PR (#53)

Comment (IceSentry) [reply kayak_core/src/color.rs:68]: Yeah, I think I'll just hardcode the red in this PR and once the color is merged I'll update this and any other place that uses hardcoded colors

File: examples/text_box.rs [hunk 2]
```diff
@@ -13,35 +15,43 @@ use kayak_ui::widgets::{App, OnChange, TextBox, Window};
 
 #[widget]
 fn TextBoxExample(context: &mut KayakContext) {
-    let value = context
-        .create_state("I started with a value!".to_string())
-        .unwrap();
-    let value2 = context.create_state("".to_string()).unwrap();
+    let (value, set_value, _) = use_state!("I started with a value!".to_string());
+    let (empty_value, set_empty_value, _) = use_state!("".to_string());
+    let (red_value, set_red_value, _) = use_state!("This text is red".to_string());
```
Comment (MrGVSV) [inline examples/text_box.rs:20]: With this change could you remove the `Bound` and `MutableBound` imports? They're automatically added by the `use_state!` macro.

Comment (IceSentry) [reply examples/text_box.rs:20]: Yep, sorry, rust-analyzer is going a bit crazy with me constantly changing branches and recompiling

Comment (MrGVSV) [reply examples/text_box.rs:20]: No worries!

**Review by MrGVSV — APPROVED** [review]

**Review by StarArawn — APPROVED** [review]

**Pull request merged**